src/DeprefixingContainer.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Dhii\Container;
+
+use Psr\Container\ContainerInterface;
+use Psr\Container\NotFoundExceptionInterface;
+
+/**
+ * A container implementation that wraps around an inner container to automatically add prefixes to keys during
+ * fetching and look up, allowing consumers to omit them.
+ *
+ * @since [*next-version*]
+ */
+class DeprefixingContainer implements ContainerInterface
+{
+    /**
+     * @since [*next-version*]
+     *
+     * @var ContainerInterface
+     */
+    protected $inner;
+
+    /**
+     * @since [*next-version*]
+     *
+     * @var string
+     */
+    protected $prefix;
+
+    /**
+     * @since [*next-version*]
+     *
+     * @var bool
+     */
+    protected $strict;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param ContainerInterface $container The container whose keys to deprefix.
+     * @param string             $prefix    The prefix to remove from the container's keys.
+     * @param bool               $strict    Whether or not to fallback to prefixed keys if an un-prefixed key does not
+     *                                      exist in the inner container.
+     */
+    public function __construct(ContainerInterface $container, string $prefix, bool $strict = true)
+    {
+        $this->inner = $container;
+        $this->prefix = $prefix;
+        $this->strict = $strict;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @since [*next-version*]
+     */
+    public function get($key)
+    {
+        try {
+            return $this->inner->get($this->getInnerKey($key));
+        } catch (NotFoundExceptionInterface $nfException) {
+            if ($this->strict) {
+                throw $nfException;
+            }
+        }
+
+        return $this->inner->get($key);
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @since [*next-version*]
+     */
+    public function has($key)
+    {
+        return $this->inner->has($this->getInnerKey($key)) || (!$this->strict && $this->inner->has($key));
+    }
+
+    /**
+     * Retrieves the key to use for the inner container.
+     *
+     * @since [*next-version*]
+     *
+     * @param string $key The outer key.
+     *
+     * @return string The inner key.
+     */
+    protected function getInnerKey($key)
```

Comment:
Is this basically the end of the `_` prefix for protected methods, due to less likely extension?

Reply:
Ah! Habit. I ditched the underscore a long time ago, never really liked it anyway. If two methods have the same name but one has an underscore, to me it indicates bad pattern. Thoughts? I have no qualms adding it back in if you deem it a necessary safety measure.

Reply:
Well, since extension is not as likely anymore, maybe we can keep it this way for now.